js/src/rive.ts

```diff
@@ -0,0 +1,965 @@
+      case Fit.Cover:
+        return rive.Fit.cover;
+      case Fit.Contain:
+        return rive.Fit.contain;
+      case Fit.Fill:
+        return rive.Fit.fill;
+      case Fit.FitWidth:
+        return rive.Fit.fitWidth;
+      case Fit.FitHeight:
+        return rive.Fit.fitHeight;
+      case Fit.ScaleDown:
+        return rive.Fit.scaleDown;
+      case Fit.None:
+      default:
+        return rive.Fit.none;
+    }
+  }
+
+  // Returns alignment for the Wasm runtime format
+  public runtimeAlignment(rive: any): any {
+    switch (this.alignment) {
+      case Alignment.TopLeft:
+        return rive.Alignment.topLeft;
+      case Alignment.TopCenter:
+        return rive.Alignment.topCenter;
+      case Alignment.TopRight:
+        return rive.Alignment.topRight;
+      case Alignment.CenterLeft:
+        return rive.Alignment.centerLeft;
+      case Alignment.CenterRight:
+        return rive.Alignment.centerRight;
+      case Alignment.BottomLeft:
+        return rive.Alignment.bottomLeft;
+      case Alignment.BottomCenter:
+        return rive.Alignment.bottomCenter;
+      case Alignment.BottomRight:
+        return rive.Alignment.bottomRight;
+      case Alignment.Center:
+      default:
+        return rive.Alignment.center;
+    }
+  }
+}
+
+// #endregion
+
+// #region runtime
+
+// Callback type when looking for a runtime instance
+export type RuntimeCallback = (rive: typeof Runtime) => void;
+
+// Runtime singleton; use getInstance to provide a callback that returns the
+// Rive runtime
+export class RuntimeLoader {
+  
+  // Singleton helpers
+  private static runtime: typeof Runtime;
+  // Flag to indicate that loading has started/completed
+  private static isLoading: boolean = false;
+  // List of callbacks for the runtime that come in while loading
+  private static callBackQueue: RuntimeCallback[] = [];
+  // Instance of the Rive runtime
+  private static rive: typeof Runtime;
+  // The url for the Wasm file
+  private static wasmWebPath: string = 'https://unpkg.com/rive-js@latest/dist/';
+  // Local path to the Wasm file; for testing purposes
+  private static wasmFilePath: string = 'dist/';
+  // Are we in test mode?
+  private static testMode: boolean = false;
+
+  // Class is never instantiated
+  private constructor() {}
+
+  // Loads the runtime
+  private static loadRuntime() : void {
+    Runtime({
+      // Loads Wasm bundle
+      locateFile: (file: string) =>
+        // if in test mode, attempts to load file locally 
+        (RuntimeLoader.testMode ?
+          RuntimeLoader.wasmFilePath :
+          RuntimeLoader.wasmWebPath) + file
+    }).then((rive: typeof Runtime) => {
+      RuntimeLoader.runtime = rive;
+      // Fire all the callbacks
+      while (RuntimeLoader.callBackQueue.length > 0) {
+        RuntimeLoader.callBackQueue.shift()(RuntimeLoader.runtime);
+      }
+    });
+  }
+
+  // Provides a runtime instance via a callback
+  public static getInstance(callback: RuntimeCallback): void {
```

Comment:
I would recommend using a Promise based API from the start instead of callback:
- it enables syntax sugar like `async` / `await`
- it provides a better integration in JS API with `Promise.all` and `Promise.allSettle` for example.
- it enables performance optimization in the browser (the method won't be blocking in the current tick as it would be flagged as asynchrone)
- it prevent you to maintain a callback API for retro compatibility reasons.

For example :
```typescript
callbackBased(cb) {
  this.queue.push(cb)
}
promiseBased() {
  return new Promise((res, rej) => this.queue.push(res));
}
```

Reply:
I've implemented both here to see how both work for users; I might get rid of the non-Promise one.